### DISCUSSION TITLE
Is it possible to add additional details to users?

### DISCUSSION THREAD

**Discussion by web-programmer-here** [nhost/nhost#77] [Q&A] [answered]
I started testing this but I have question about user management. Is it possible to add more details to users? like Name, Last Name, Phone?

**Comment by elitan** [comment]
Yes, you can add columns to the users' table.

Just note that we're working on Nhost v2 (https://waitlist.nhost.io/) where adding columns to the users' table will not be allowed. Instead, we'll have a `metadata` column where you can add custom JSON data to any user.

So if you want to "future-proof" your app, I would recommend adding a single `metadata` column to the `users` table with a JSONB datatype.

**Comment by elitan** [comment, marked as answer]
So, since we released Nhost 2.0 the official answer is: No. Do not edit the `auth.users` table.

What you can do instead is create a `public.profiles` table and link the two tables to each other.

We also plan to support a generic `metadata` column on the `auth.users` table, tracked here: https://github.com/nhost/hasura-auth/issues/31

**Reply by brentrobbins** [reply]
@elitan I created a public.profiles table like suggested, but is it OK to add the relationship on the auth.users table to the new public.profiles table? I know you mentioned not to change that table.

**Reply by elitan** [reply]
Yes. Adding custom relationships and permissions are totally fine to do for both all tables in the `auth` and `storage` schemas.

**Reply by brentrobbins** [reply]
Great! I figured, but I wanted to make sure. Thanks for letting me know.

**Reply by st3phan** [reply]
What I don't like about this is that you currently have to do an extra query in the signUp process. And it's hard to make it transactional in case the signUp or the insertProfile fails. So I prefer to save all user data in one go.
So https://github.com/nhost/hasura-auth/issues/31 is very high on my wishlist ;-)

Of course you can still have the profiles table that gets populated with and hasura event that reads from `metadata` column that will be added in https://github.com/nhost/hasura-auth/issues/31.
At least that's how I image it would work.

**Reply by sangdth** [reply]
Sorry to dig this up. But after a while working with `profiles` table, I can say it's too much labor tasks for developer. A simple action like update move this user from this tenant/organization to another one requires so many query and mutation.

I hope we can find a better solution.

**Reply by elitan** [reply]
We're working on making it easier to add custom claims and add custom data to users. Check this: https://github.com/nhost/hasura-auth/pull/78
